stop-gateway.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+# Stop all services and optionally remove specified volumes to remove all state and start fresh
+
+# Export the variables that are used in the docker-compose.yaml file
+if [ -f .env-saved ]; then
+    set -a; source .env-saved; set +a
+fi
+
+# Shutting down any running services
+echo "Shutting down any running services..."
+docker compose --profile local-node --profile backend --profile frontend down
+
+# Ask the user if they want to remove specified volumes
+echo "Do you want to remove specified volumes to remove all state and start fresh? [y/n]: "
+read REMOVE_VOLUMES
+
+if [[ $REMOVE_VOLUMES =~ ^[Yy]$ ]]
+then
+    echo "Removing specified volumes..."
+    # Docker volume names are lowercase versions of the directory name
+    # In the root directory of the repository, we get from the system directory name
+    docker volume rm $(basename "$(pwd)" | tr '[:upper:]' '[:lower:]')_redis_data
```

Comment:
You can use the `--project-name` argument to `docker compose` to override the default directory-based name of the project, so you don't have to do all of the `$(basename ...)` garbage